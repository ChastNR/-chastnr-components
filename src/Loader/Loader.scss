.ldr {
  flex-shrink: 0;
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  text-align: center;
  vertical-align: bottom;

  .spnr {
    color: rgb(255, 221, 45);
    display: inline-block;
    height: 56px;
    width: 100%;
  }

  &.fs {
    height: 100vh;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background: rgba(51, 51, 51, 0.3);

    .spnr {
      height: 72px;
    }
  }
}
