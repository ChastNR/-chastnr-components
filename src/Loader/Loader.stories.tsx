/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta } from '@storybook/react';
import { Loader } from './Loader';

const componentMeta: ComponentMeta<typeof Loader> = {
  title: 'Loader',
  component: Loader,
  parameters: {
    layout: 'fullscreen',
  },
};

export const DefaultLoader = () => <Loader />;

export default componentMeta;
