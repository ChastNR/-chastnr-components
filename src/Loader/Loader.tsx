import classNames from 'clsx';
import { Spinner } from '../base';

import './Loader.scss';

interface ILoaderProps {
  className?: string;
  fullScreen?: boolean;
}

export const Loader: React.FC<ILoaderProps> = ({ className, fullScreen }) => {
  const loaderClassName = classNames('ldr', className, {
    fs: fullScreen,
  });

  return (
    <div className={loaderClassName}>
      <Spinner />
    </div>
  );
};
