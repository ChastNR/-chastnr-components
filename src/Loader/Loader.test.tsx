import { render } from '@testing-library/react';
import { Loader } from './Loader';

describe('Loader', () => {
  it('should render properly', () => {
    const loader = render(<Loader />);

    expect(loader.container.firstChild).toMatchSnapshot();
  });
});
