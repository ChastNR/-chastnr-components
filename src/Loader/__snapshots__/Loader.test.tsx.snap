// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render properly 1`] = `
<div
  class="ldr"
>
  <span
    class="spnr"
  >
    <span>
      content
    </span>
  </span>
</div>
`;
